src/ir/cost.h

```diff
@@ -678,6 +678,9 @@ struct CostAnalyzer : public OverriddenVisitor<CostAnalyzer, CostType> {
   CostType visitStringMeasure(StringMeasure* curr) {
     return 6 + visit(curr->ref);
   }
+  CostType visitStringEncode(StringEncode* curr) {
+    return 6 + visit(curr->ref) + visit(curr->ptr);
```

Comment:
I guess cost for encoding should be bigger than `visitArrayCopy` has. Also it should depends on format. `encode_wtf16` probably will have slightly less cost than `encode_wtf8`

Reply:
Got it!